spring-session-data-redis/src/main/java/org/springframework/session/data/redis/RedisIndexedSessionRepository.java

```diff
@@ -489,7 +493,12 @@ public void deleteById(String sessionId) {
 
 	@Override
 	public RedisSession createSession() {
-		MapSession cached = new MapSession();
+		return createSession(null);
+	}
+
+	@Override
+	public RedisSession createSession(final String id) {
+		MapSession cached = Optional.ofNullable(id).map(MapSession::new).orElse(new MapSession());
```

Comment:
It's better to use `.orElseGet(MapSession::new)` to avoid creating an unnecessary object if `id` is provided.